extensions/functions_arithmetic_decimal.yaml

```diff
@@ -149,3 +149,19 @@ aggregate_functions:
         decomposable: MANY
         intermediate: "DECIMAL?<P, S>"
         return: "DECIMAL?<P, S>"
+  - name: "sum0"
+    description: >
+      Sum a set of values. The sum of zero elements yields zero.
+
+      Null values are ignored.
+    impls:
+      - args:
+          - name: x
+            value: "DECIMAL<P, S>"
+        options:
+          overflow:
+            values: [ SILENT, SATURATE, ERROR ]
+        nullability: DECLARED_OUTPUT
+        decomposable: MANY
+        intermediate: "DECIMAL<38,S>"
+        return: "DECIMAL<38,S>"
```

Comment:
Should this return type be `DECIMAL<P,S>`? Does `sum0` change the precision somehow?

Reply:
> Should this return type be `DECIMAL<P,S>`? Does `sum0` change the precision somehow?

I think this behavior should be the same as sum, always plan will do cast later.

Reply:
It appears that any operation where decimals aren't manipulated by each other return the same precision.  Any time it does, we are returning the max precision (as we do here).  It doesn't seem likely that we'd increase precision by doing a lot of add operations though.  However it is consistent with sum (and the other similar arithmetic operations).  We can address the correctness for all of them together I suppose.